BootloaderCommonPkg/Library/TpmLib/TpmLib.c

```diff
@@ -611,14 +611,46 @@ TpmExtendStageHash (
 {
   RETURN_STATUS        Status;
   CONST UINT8         *Digest;
+  UINT8                HashAlg;
 
   // Get public key hash
-  Status = GetComponentHash (ComponentType, &Digest);
+  Status = GetComponentHash (ComponentType, &Digest, &HashAlg);
   if(Status == EFI_SUCCESS) {
     Status = TpmExtendPcrAndLogEvent ( 0, TPM_ALG_SHA256, Digest,
       EV_POST_CODE, POST_CODE_STR_LEN, (UINT8 *)EV_POSTCODE_INFO_POST_CODE);
   } else {
     DEBUG ((DEBUG_ERROR, "Error: Component (%d) is not measured in TPM.\n", ComponentType));
   }
+
+  return Status;
+}
+
+
+/**
+  Extend  hash provided in PCR 0 with EV_POST_CODE event type.
+
+  @param[in] Digest                    Hash Digest to extend
+  @param[in] HashAlg                   Tcg Alg to extend
+
+  @retval RETURN_SUCCESS               Operation completed successfully.
+  @retval RETURN_INVALID_PARAMETER     Imvalid Digest  buf
+
+  @retval Others                       Unable to extend stage hash.
+**/
+RETURN_STATUS
+TpmExtendHash (
+  IN       UINT8            *Digest,
+  IN       UINT16            HashAlg

```

Comment:
do we need UINT16 ?  or UINT8 ?

Reply:
We expect hash values defined in Cryptolib.h which are NOT bitmasks. UINT8 would be okay.
#define  HASH_TYPE_SHA256              1
#define  HASH_TYPE_SHA384              2
#define  HASH_TYPE_SHA512              3
#define  HASH_TYPE_SM3                 4

Reply:
let us use UINT8  then.

Reply:
For TPM library interface, we are using variables from IndustryStandard\Tpm20.h file.
Hash is referred by TPMI_ALG_HASH variable in the existing methods in TpmLib.h
We should use that,

Reply:
SBL crypto hash would be available in stage components through Hash store/container headers. My initial thought was to add a hash convert from SBL crypto hash to TPMI_ALG_HASH inside TpmExtendHash. 
 I would a convert to TPMI_ALG_HASH and pass it.